onnxruntime/contrib_ops/cuda/bert/group_query_attention_helper.h

```diff
@@ -240,10 +239,25 @@ Status CheckInputs(const Tensor* query,
                            "Input 'cos_cache' and 'sin_cache' shall be both present or both absent.");
   }
 
-  bool is_prompt = (sequence_length == total_sequence_length);
-  if (!is_prompt && sequence_length != 1) {
-    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
-                           "sequence_length shall be 1 when it is not prompt.");
+  bool is_subsequent_prompt = false;
```

Comment:
This logic seems duplicated with cpu/bert/group_query_attention_helper.h. Can we consolidate these two header files into one?